### PR TITLE
Don't pass a Fiber to showErrorDialog()

### PR DESCRIPTION
**Pull request by gaearon**
There are two places where we create CapturedError. The one in reconciliation phase correctly passes the `stateNode` but we accidentally broke it for the commit phase. This should fix it. To reduce duplication (which confused me at first), I removed the `else` branch which was already taken care of with the default values.

### CODE REVIEW

File: packages/react-reconciler/src/ReactFiberCommitWork.js [hunk 1]
```diff
@@ -54,20 +54,18 @@ function logError(boundary: Fiber, errorInfo: CapturedValue<mixed>) {
...
     componentStack: stack !== null ? stack : '',
     errorBoundaryName: null,
     errorBoundaryFound: false,
     willRetry: false,
   };
 
   if (boundary !== null) {
+    capturedError.errorBoundary = boundary.stateNode;
     capturedError.errorBoundaryName = getComponentName(boundary);
     capturedError.errorBoundaryFound = capturedError.willRetry =
```
Comment (gaearon) [inline packages/react-reconciler/src/ReactFiberCommitWork.js]: This line still confuses me a bit. Can it be something other than `ClassComponent`? Should the `errorBoundary` assignment also be conditional on that?

**Comment by bvaughn** [comment]
Maybe we should add a test?

File: packages/react-reconciler/src/ReactFiberCommitWork.js (hunk 1 above)
Comment (bvaughn) [reply packages/react-reconciler/src/ReactFiberCommitWork.js]: `componentDidCatch` only works for class component API.

**Comment by gaearon** [comment]
I'm already leaving but happy if somebody takes over

File: packages/react-reconciler/src/ReactFiberCommitWork.js (hunk 1 above)
Comment (gaearon) [reply packages/react-reconciler/src/ReactFiberCommitWork.js]: Right, which is why the `tag` check on the next line confused me.

Comment (acdlite) [reply packages/react-reconciler/src/ReactFiberCommitWork.js]: https://github.com/gaearon/react/blob/509b257a30839b6b47f53ae522679756161cdf84/packages/react-reconciler/src/ReactFiberCommitWork.js#L291

**Comment by gaearon** [comment]
@acdlite does the update make sense to you?

**Comment by acdlite** [comment]
@gaearon Yep! Eventually we should clean this up and formalize a more official API for error logging.

**Pull request merged**